data_structures/linked_list.cpp

```diff
@@ -1,58 +1,147 @@
+/**
+ * @file linked_list.cpp
+ * @brief Implementation of singly linked list algorithm.
+ *
+ * @details
+ * The linked list is a form of container used for holding a sequence of 
+ * values, which can be added, removed and displayed.
+ * ### Algorithm
+ * Values can be added by iterating to the end of a list(by following 
+ * the pointers) starting from the first node. Whichever node points to null
+ * is considered the last node and is pointed to the new value.
+ *
+ * Values can be removed by also iterating through the list. When the node containing 
+ * the value is found, the node pointing to the current node is made to point to the 
+ * node that the current node is pointing to, and then returning the current node to
+ * heap store.
+ */
 #include <iostream>
+#include <sstream>
 
-struct node {
+/**
+ * @namespace data_structures
+ * @brief Data Structures algorithms
+ */
+namespace data_structures {
+/**
+ * Class Documentation
```

Comment:
Please add proper documentation.
It doesn't say how it works, what it does, etc..